crates/wasi-common/src/hostcalls/sock.rs

```diff
@@ -41,3 +43,25 @@ pub unsafe fn sock_shutdown(
 ) -> wasi::__wasi_errno_t {
     unimplemented!("sock_shutdown")
 }
+
+hostcalls! {
+    pub unsafe fn sock_connect(
+        wasi_ctx: &WasiCtx,
+        memory: &mut [u8],
+        sock: wasi::__wasi_fd_t,
+        addr_ptr: wasi32::uintptr_t,
+        addr_len: wasi32::size_t,
+    ) -> wasi::__wasi_errno_t;
+
+    pub unsafe fn sock_socket(
+        wasi_ctx: &mut WasiCtx,
+        memory: &mut [u8],
+        sock_domain: i32,
+        // socket type
+        // DGRAM 5
+        // STREAM 6
+        sock_type: wasi::__wasi_filetype_t,
+        sock_protocol: i32,
```

Comment:
I presume that the `i32`s will be converted into the actual WASI types, right?

Reply:
There was a `wasm32::int32_t` type before, but now I can't find it. What should be an equivalent for i32 from WASI now? And where can I find all the available types? It looks like they're autogenerated

Reply:
You're right, I can't see it in `wasi32` now either. Perhaps we should introduce some additional typenames specifically for socket's domain and protocol? Perhaps some enums even? Would that make sense?